src/main/java/com/aws/greengrass/logmanager/CloudWatchAttemptLogsProcessor.java

```diff
@@ -330,6 +330,10 @@ private boolean addNewLogEvent(AtomicInteger totalBytesRead, CloudWatchAttemptLo
             return true;
         }
 
+        // Cloudwatch does not allow uploading data older than 14 days.
+        if (timestamp.isBefore(Instant.now(clock).minus(14, ChronoUnit.DAYS))) {
+            return false;
+        }
```

Comment:
It still seems like this can return true even if the timestamp is too early. I haven't gone through the log manager code so - does that matter that this can return that the maximum message size has been reached even if the message to be added is filtered out?

Reply:
Returning true/false here doesn't mean that the log event will be uploaded, it is actually signaling if the batch is full or not. Adding the event happens at the very end of this method. It is not possible that a log event which is too old would be added

Reply:
right - I can see that we don't add the event - we return.
I'm just curious about the logic of this flag - if `current + x > limit`, but we aren't going to use `x`, why are we even checking?

Reply:
You're talking about L329? We want to check if we were to add the event, would it be too big. If so, then the batch is too big and this current event needs to go into the next batch.

Reply:
ok - it seems a bit odd to say the batch would be too large when the message that would trigger it being too large isn't ever going to be added to the batch. 

This is a bit of an edge case though so I'm not too worried. I'm just trying to understand what this code is doing

Reply:
The current event cannot fit, therefore the batch cannot be added to anymore. We don't try any further log events to try and fill the batch 100% or anything like that.